chia/util/condition_tools.py

```diff
@@ -71,6 +71,45 @@ def agg_sig_additional_data(agg_sig_data: bytes) -> Dict[ConditionOpcode, bytes]
     return ret
 
 
+def make_aggsig_final_message(
+    opcode: ConditionOpcode,
+    msg: bytes,
+    spend: Union[Coin, Spend],
+    agg_sig_additional_data: Dict[ConditionOpcode, bytes],
+) -> bytes:
+    if isinstance(spend, Coin):
+        parent_coin_info = bytes(spend.parent_coin_info)
+        puzzle_hash = bytes(spend.puzzle_hash)
+        amount = int_to_bytes(spend.amount)
+        coin_id = bytes(spend.name())
+    elif isinstance(spend, Spend):
```

Comment:
You could simply build a `Coin` out of the `Spend` here and then your logic could always assume `Coin` type.

Reply:
Thanks, done.